sdk/file_sdk/XMPFileFormat/xmpio_over_istream.cpp

```diff
@@ -60,20 +61,28 @@ void XMPIOOverIStream::Truncate(XMP_Int64 length) {
 }
 
 XMP_IO* XMPIOOverIStream::DeriveTemp() {
+  if(mTemp)
+    return mTemp;
   auto backingStdStream = make_shared<std::stringstream>(std::ios::in | std::ios::out | std::ios::binary);
   auto stream = StdStreamAdapter::Create(static_pointer_cast<std::iostream>(backingStdStream));
-  delete mTemp;
   mTemp = new XMPIOOverIStream(stream);
   return mTemp;
 }
 
 void XMPIOOverIStream::AbsorbTemp() {
   if (mTemp == nullptr)
     return;
-  mBaseStream->Seek(0);
   mTemp->mBaseStream->Seek(0);

```

Comment:
destroy mTemp?

Reply:
Do you mean:
void XMPIOOverIStream::DeleteTemp() {
  delete mTemp;
  mTemp = nullptr;
}

Reply:
Yeah. NVM.